activerecord/lib/active_record/database_configurations.rb

```diff
@@ -248,7 +248,7 @@ def build_db_config_from_string(env_name, name, config)
       end
 
       def build_db_config_from_hash(env_name, name, config)
-        if config.has_key?(:url)
+        if config.has_key?(:url) && config[:url].present?
```

Comment:
This can be simplified

```suggestion
        if config[:url].present?
```

If there's no key `:url`, `config[:url].present?` is going to return `false` anyway.

Reply:
you are right, thank you. Fixed!

Reply:
wow, only now i realised that there was an option to commit it directly from github! so cool :)

Reply:
@DNNX can u take another look, pls?

Reply:
It's beautiful!

If you want me to merge the pull request - you better ask someone else. I'm just a bypasser, I do not have any permissions on this repo.